catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/QueryApplication.java

```diff
@@ -74,7 +86,25 @@ public void init() {
 
         after("/cql", (req, res) -> {
             res.type(APPLICATION_JSON);
-            res.header("Content-Encoding", "gzip");
+
+            // Must manually check and set header for gzip because of spark issue
+            // https://github.com/perwendel/spark/issues/691
+            Map<String, String> acceptEncodings =
+                    parseAcceptEncodings(req.headers("Accept-Encoding"));
+
+            if (acceptEncodingSupports(acceptEncodings, "gzip")) {
+                res.header("Content-Encoding", "gzip");
+            } else if (acceptEncodingSupports(acceptEncodings, "identity")) {
```

Comment:
Technically `identity` could have a higher `qvalue` than `gzip` which would make this `else if` invalid.  Not sure why anyone would want to do that though.